### PR TITLE
Optionally format s3_path_dimensions using os.date

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: parquet/sandboxes/heka/output/s3_parquet.lua [hunk 1]
```diff
@@ -112,6 +114,8 @@ require "parquet"
 local load_schema = require "lpeg.parquet".load_parquet_schema
 require "string"
 require "table"
+local date          = require "os".date
+local floor         = require "math".floor
```
Comment (trink) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: You don't need floor in this case

File: parquet/sandboxes/heka/output/s3_parquet.lua [hunk 2]
```diff
@@ -242,6 +246,9 @@ local function get_s3_path(json)
         else
             v = read_json(json, d.source) or default_nil
         end
+        if d.dateformat and type(v) == "number" then
```
Comment (trink) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: So this allows any field to contain dateformat but requires it is in nanoseconds.  We could either restrict it to Timestamp only and dump the number test or we allow a scaling factor (say there is a time_t in some other field)

Comment (relud) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: scaling factor for sure

File: parquet/sandboxes/heka/output/s3_parquet.lua (hunk 1 above)
Comment (relud) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: I'm using it on line 250, and without it os.date() will throw the error: `bad argument #2 to 'date' (number has no integer representation)`

File: parquet/sandboxes/heka/output/s3_parquet.lua (hunk 2 above)
Comment (relud) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: I'll add a scaling factor that defaults to nanoseconds

File: parquet/sandboxes/heka/output/s3_parquet.lua (hunk 1 above)
Comment (trink) [inline parquet/sandboxes/heka/output/s3_parquet.lua]: line 120 can be removed along with the floor call on 252

**Pull request merged**